packages/langium/src/lsp/completion/completion-provider.ts

```diff
@@ -381,28 +376,26 @@ export class DefaultCompletionProvider implements CompletionProvider {
         const assignment = getContainerOfType(crossRef.feature, ast.isAssignment);
         let node = context.node;
         if (assignment && node) {
-            if (crossRef.type && (crossRef.new || node.$type !== crossRef.type)) {
+            if (crossRef.type) {
                 node = {
                     $type: crossRef.type,
                     $container: node,
                     $containerProperty: crossRef.property
                 };
-            }
-            if (!context) {
-                return;
+                assignMandatoryAstProperties(this.astReflection, node);
             }
             const refInfo: ReferenceInfo = {
-                reference: {} as Reference,
+                reference: {
+                    $refText: ''
+                },
                 container: node,
                 property: assignment.feature
             };
             try {
                 const scope = this.scopeProvider.getScope(refInfo);
-                scope.getAllElements().forEach(e => {
-                    if (this.filterCrossReference(context, e)) {
-                        acceptor(context, this.createReferenceCompletionItem(e));
-                    }
-                });
+                for (const description of this.filterCrossReferences(context, scope.getAllElements())) {
+                    acceptor(context, this.createReferenceCompletionItem(description));
+                }
```

Comment:
This is just a different way of formulating the same, correct? (except for the changed signature of `filterCrossReferences(...)`)

Reply:
Kind of yes. I just like the `for` loop more, since it's easier to debug.